src/Symfony/Component/DependencyInjection/Dumper/XmlDumper.php

```diff
@@ -137,8 +137,14 @@ private function addService(Definition $definition, ?string $id, \DOMElement $pa
                 } else {
                     $tag->appendChild($this->document->createTextNode($name));
                 }
-                foreach ($attributes as $key => $value) {
-                    $tag->setAttribute($key, $value ?? '');
+
+                // Check if we have recursive attributes
+                if (array_filter($attributes, static fn ($v) => \is_array($v))) {
```

Comment:
the fn should be replaced by `\is_array(...)`

Reply:
hah, sorry, I didn't know about that syntax 😆